src/WooCommerce.php

```diff
@@ -243,7 +243,7 @@ public static function getProductVariationsImages($product_id) {
     $results = array_map('maybe_unserialize',
       $wpdb->get_col($wpdb->prepare(
         "
-        SELECT pm.meta_value
+        SELECT DISTINCT pm.meta_value
         FROM wp_postmeta pm
         INNER JOIN wp_posts p ON p.ID = pm.post_id AND p.post_parent = %d
         WHERE pm.meta_key IN ('_thumbnail_id', '_gallerya_attachment_ids')
```

Comment:
About your comment about duplicated variation images on the parent product. Does it make sense to look into also excluding the images attached to the parent product? I'm not sure how the variation image galleries work so this suggestion might not be useful.

Maybe we can exclude them here using something like `WHERE media.id NOT IN(1,2,3,4)` and parse in the image IDs attached to the parent product.

If the post IDs are not available then how about a sub query?  `WHERE id NOT IN (SELECT ids FROM media)`